discovery-provider/src/tasks/index_aggregate_track.py

```diff
@@ -0,0 +1,215 @@
+    SELECT
+        DISTINCT(t.track_id),
+        COALESCE(track_repost.repost_count, 0) AS repost_count,
+        COALESCE(track_save.save_count, 0) AS save_count
+    FROM
+        t AS (
+            SELECT
+                t.track_id
+            FROM
+                tracks t
+            WHERE
+                t.is_current IS TRUE
+                AND t.is_delete IS FALSE
+                AND t.is_unlisted IS FALSE
+                AND t.stem_of IS NULL
+            GROUP BY
+                t.track_id
+        )
+        LEFT OUTER JOIN (
+            SELECT
+                r.repost_item_id AS track_id,
+                count(r.repost_item_id) AS repost_count
+            FROM
+                reposts r
+            WHERE
+                r.is_current IS TRUE
+                AND r.repost_type = 'track'
+                AND r.is_delete IS FALSE
+                AND r.repost_item_id IN (
+                    SELECT
+                        track_id
+                    FROM
+                        new_track
+                )
+                AND r.blocknumber > (
+                    SELECT
+                        prev_blocknumber
+                    FROM
+                        aggregate_track_checkpoints
+                )
+                AND r.blocknumber <= (
+                    SELECT
+                        current_blocknumber
+                    FROM
+                        aggregate_track_checkpoints
+                )
+            GROUP BY
+                r.repost_item_id
+        ) AS track_repost ON track_repost.track_id = t.track_id
+        LEFT OUTER JOIN (
+            SELECT
+                s.save_item_id AS track_id,
+                count(s.save_item_id) AS save_count
+            FROM
+                saves s
+            WHERE
+                s.is_current IS TRUE
+                AND s.repost_type = 'track'
+                AND s.is_delete IS FALSE
+                AND s.save_item_id IN (
+                    SELECT
+                        track_id
+                    FROM
+                        new_track
+                )
+                AND s.blocknumber > (
+                    SELECT
+                        prev_blocknumber
+                    FROM
+                        aggregate_track_checkpoints
+                )
+                AND s.blocknumber <= (
+                    SELECT
+                        current_blocknumber
+                    FROM
+                        aggregate_track_checkpoints
+                )
+            GROUP BY
+                s.save_item_id
+        ) AS track_save ON track_save.user_id = t.track_id
+    WHERE
+        t.track_id in (
+            SELECT
+                track_id
+            FROM
+                new_track
+        ) ON CONFLICT (track_id) DO
+    UPDATE
+    SET
+        repost_count = aggregate_track.repost_count + EXCLUDED.repost_count
+        save_count = aggregate_track.save_count + EXCLUDED.save_count
```

Comment:
I don't think it should only add. What if repost / save counts actually go down.

Reply:
If the counts go down then it would still add that negative number to the existing sum would would still be a subtraction. Thoughts?

Reply:
Transferred the aggregate_user tests into this PR for aggregate_track, except the tests centered around truncating the table.

Let's chat about this on Monday to get your thoughts, but this PR is pretty much done now!

Reply:
Our chat on Monday led us to recalculate the total from 0, not use a running total for now.